packages/api/src/brevo-contact/brevo-contact-import.service.ts

```diff
@@ -0,0 +1,169 @@
+import { parseString } from "@fast-csv/parse";
+import { Inject, Injectable } from "@nestjs/common";
+import { IsEmail, IsNotEmpty, validateOrReject } from "class-validator";
+
+import { isErrorFromBrevo } from "../brevo-api/brevo-api.utils";
+import { BrevoApiContactsService, CreateDoubleOptInContactData } from "../brevo-api/brevo-api-contact.service";
+import { BrevoContactsService } from "../brevo-contact/brevo-contacts.service";
+import { BrevoModuleConfig } from "../config/brevo-module.config";
+import { BREVO_MODULE_CONFIG } from "../config/brevo-module.constants";
+import { TargetGroupInterface } from "../target-group/entity/target-group-entity.factory";
+import { TargetGroupsService } from "../target-group/target-groups.service";
+import { EmailCampaignScopeInterface } from "../types";
+
+class ValidateableRow {
+    @IsEmail()
+    @IsNotEmpty()
+    email: string;
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any;
+}
+
+@Injectable()
+export class BrevoContactImportService {
+    constructor(
+        @Inject(BREVO_MODULE_CONFIG) private readonly config: BrevoModuleConfig,
+        private readonly brevoApiContactsService: BrevoApiContactsService,
+        private readonly brevoContactsService: BrevoContactsService,
+        private readonly targetGroupsService: TargetGroupsService,
+    ) {}
+
+    async importContactsFromCsv<T extends ValidateableRow>(
+        csvContent: string,
+        scope: EmailCampaignScopeInterface,
+        redirectUrl: string,
+        targetGroups: TargetGroupInterface[] = [],
+        validateable?: T,
```

Comment:
is this ever defined ?

I can't find the function with 5 parameters, so it should be always undefined, or am I missing something?

Reply:
This was added by @thomashuettmaier. I guess it was meant for some kind of more advanced validation logic. But apparently it was never used. I don't think we need it for now, so I removed it. We can maybe improve validation in the future, if needed

Reply:
yes I think we should probably improve the validation in the future, especially for custom user uploads in the csv.